OpenRA.Game/WorldUtils.cs

```diff
@@ -41,6 +41,29 @@ public static IEnumerable<Actor> FindActorsInCircle(this World world, WPos origi
 			}
 		}
 
+		public static bool ContainsTemporaryBlocker(this World world, CPos cell, Actor ignoreActor = null)
+		{
+			var temporaryBlockers = world.ActorMap.GetActorsAt(cell);
+			if (temporaryBlockers == null)
+				return false;
+
+			foreach (var temporaryBlocker in temporaryBlockers)
+			{
+				if (temporaryBlocker == ignoreActor)
+					return false;
+
+				var temporaryBlockerTraits = temporaryBlocker.TraitsImplementing<ITemporaryBlocker>();
+				if (temporaryBlockerTraits == null)
+					return false;
+
+				foreach (var temporaryBlockerTrait in temporaryBlockerTraits)
+					if (temporaryBlockerTrait.IsBlocking(temporaryBlocker, cell))
+						return true;
```

Comment:
This wants to stay as `return true;` since you know know that there definitely is a blocker.